path.py

```diff
@@ -0,0 +1,167 @@
+        with open(str(pathName)) as f:
+            for index, line in enumerate(f):
+                self.executeSegment(line.strip())
+
+    def executeSegment(self, line):			#if adding checkpoint with arm movement, can pass segment[1] in executeDirection
+        segment = line.split()				#segment[1] would only be used in direction == checkpoint
+        self.executeDirection(segment[0])
+        time.sleep(int(float(segment[1])))
+        self.navigate.stop()
+
+    def executeDirection(self, direction):
+        if direction == "forward":
+            self.navigate.forward()
+        elif direction == "backward":
+            self.navigate.backward()
+        elif direction == "right":
+            self.navigate.right()
+        elif direction == "left":
+            self.navigate.left()
+        elif direction == "CW":
+            self.navigate.cw()
+        elif direction == "CCW":
+            self.navigate.ccw()
+#        elif direction == "checkpoint":
+#            do arm movement at designated aruco_id
+
+    def recordPath(self, pathName):
+        self.pathFile = FileManagement(pathName)
+        self.pathFile.writeLine("start", "0")           #to set checkpoint it would be "set checkpoint command" sent by app - TBD
+        data = self.BLE.read()
+
+        while data != f'{Commands.END_TASK.value}'.encode():
+            self.recordSegment(data)
+            data = self.BLE.read()
+#            if data == b'c\r\n':
+#                self.setCheckpoint()
+
+        self.atHomeBase()
+
+    def recordSegment(self, data):
+        direction = self.getDirection(data)
+        travelTime = self.getTime()
+        self.pathFile.writeLine(direction, travelTime)
+        self.numLines += 1
+
+    def atHomeBase(self):
+#        rvec, tvec = aruco.estimatePose()
+#        if not rvec and not tvec:
+#            print("No aruco marker found. Reversing path back to home base.")
+        self.reversePath()
+        self.pathFile.writeFile("end", "0")        #no aruco_id because path was reversed
+#        else:
+#            self.pathFile.writeFile("end", self.aruco_id)
+
+#    def setCheckpoint(self):
+#        rvec, tvec = aruco.estimatePose()
+#        if not rvec and not tvec:
+#            print("Error: no aruco marker found. Can't set checkpoint here")
+#        else:
+#            self.pathFile.writeLine("checkpoint", self.aruco_id)
+#            self.aruco_id += 1
+
+    def getTime(self):                                  #times seem a bit off - check
+        startTime = datetime.now()
+
+        isStop = self.BLE.read()
+
+        assert isStop == f'{Commonds.STOP.value}'.encode()
+
+        self.navigate.stop()
+        
+        endTime = datetime.now()
+        timeString = endTime - startTime
+        return timeString.total_seconds()
+
+    def getDirection(self, direction):
+        if direction == f'{Commands.FORWARD.value}'.encode():
+            self.navigate.forward()
+            return "forward"
```

Comment:
it'd be better to have a `Direction` enum for these constants

Reply:
I'll add it to my to do list and merge it with the changes that I'll make to the path file after navigation testing.